code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -171,7 +171,10 @@
 	var/ex_heavy = round(created_volume / 42)
 	var/ex_light = round(created_volume / 20)
 	var/ex_flash = round(created_volume / 8)
-	explosion(location, ex_severe, ex_heavy,ex_light, ex_flash, 1)
+	if(istype(holder.my_atom))
```

Comment:
Can you add a comment to this why this is done?
 Also might want to rename `location` to `prime_location` so it is a bit less confusing

Reply:
applied requested changes